cmd/state/exec3/historical_trace_worker.go

```diff
@@ -420,7 +405,7 @@ func CustomTraceMapReduce(fromBlock, toBlock uint64, consumer TraceConsumer, ctx
 	}
 	logEvery := time.NewTicker(1 * time.Second)
 	defer logEvery.Stop()
-	for blockNum := fromBlock; blockNum <= toBlock; blockNum++ {
+	for blockNum := fromBlock; blockNum < toBlock && !workersExited.Load(); blockNum++ {
```

Comment:
it's because [from, to) but we've decreased toBlock already.. would it be a problem?

Reply:
don't know - such corner-cases can only be checked by unit-tests. changed to `<=`